static/js/components/tiles/donut_tile.tsx

```diff
@@ -0,0 +1,238 @@
+  statVarSpec: StatVarSpec[];
+  // Height, in px, for the SVG chart.
+  svgChartHeight: number;
+  // Extra classes to add to the container.
+  className?: string;
+  // API root
+  apiRoot?: string;
+}
+
+interface DonutChartData {
+  dataGroup: DataGroup[];
+  sources: Set<string>;
+  unit: string;
+  dateRange: string;
+}
+
+export function DonutTile(props: DonutTilePropType): JSX.Element {
+  const chartContainerRef = useRef<HTMLDivElement>(null);
+  const [donutChartData, setDonutChartData] = useState<
+    DonutChartData | undefined
+  >(null);
+
+  useEffect(() => {
+    if (!donutChartData) {
+      (async () => {
+        const data = await fetchData(props);
+        setDonutChartData(data);
+      })();
+    }
+  }, [props, donutChartData]);
+
+  const drawFn = useCallback(() => {
+    if (_.isEmpty(donutChartData)) {
+      return;
+    }
+    draw(props, donutChartData, chartContainerRef.current);
+  }, [props, donutChartData]);
+
+  useDrawOnResize(drawFn, chartContainerRef.current);
+
+  return (
+    <ChartTileContainer
+      id={props.id}
+      title={props.title}
+      sources={donutChartData && donutChartData.sources}
+      replacementStrings={getReplacementStrings(props, donutChartData)}
+      className={`${props.className} bar-chart`}
+      allowEmbed={true}
+      getDataCsv={
+        donutChartData ? () => dataGroupsToCsv(donutChartData.dataGroup) : null
+      }
+      isInitialLoading={_.isNull(donutChartData)}
+    >
+      <div
+        id={props.id}
+        className="svg-container"
+        style={{ minHeight: props.svgChartHeight }}
+        ref={chartContainerRef}
+      ></div>
+    </ChartTileContainer>
+  );
+}
+
+// Get the ReplacementStrings object used for formatting the title
+export function getReplacementStrings(
+  props: DonutTilePropType,
+  chartData: DonutChartData
+): ReplacementStrings {
+  return {
+    placeName: props.place ? props.place.name : "",
+    date: chartData && chartData.dateRange,
+  };
+}
+
+export const fetchData = async (props: DonutTilePropType) => {
+  const statVars = [];
+  for (const spec of props.statVarSpec) {
+    statVars.push(spec.statVar);
+    if (spec.denom) {
+      statVars.push(spec.denom);
+    }
+  }
+  // Fetch populations.
+  statVars.push(FILTER_STAT_VAR);
+  const url = `${props.apiRoot || ""}/api/observations/point`;
+  const params = {
+    entities: [props.place.dcid],
+    variables: statVars,
+  };
+  try {
+    const resp = await axios.get<PointApiResponse>(url, {
+      params,
+      paramsSerializer: stringifyFn,
+    });
+
+    // Find the most populated places.
+    let popPoints: RankingPoint[] = [];
+    for (const place in resp.data.data[FILTER_STAT_VAR]) {
+      popPoints.push({
+        placeDcid: place,
+        value: resp.data.data[FILTER_STAT_VAR][place].value,
+      });
+    }
+    // Take the most populated places.
+    popPoints.sort((a, b) => a.value - b.value);
+    popPoints = popPoints.slice(0, NUM_PLACES);
```

Comment:
I ended up refactoring this logic in #2903 - can you let me know what you think & align approaches in a follow-up pr?

Reply:
Love the refactor -- will match the approach in a followup PR.